apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/OrderRow/index.tsx

```diff
@@ -173,10 +171,10 @@ export function OrderRow({
   }, [orderActions, order])
   const alternativeOrderModalContext = useMemo(
     () => orderActions.getAlternativeOrderModalContext(order),
-    [order, orderActions]
+    [order, orderActions],
   )
 
-  const withAllowanceWarning = hasEnoughAllowance === false && hasValidPendingPermit === false
+  const withAllowanceWarning = hasEnoughAllowance === false
```

Comment:
Now it's simplier, `hasEnoughAllowance` already includes result of permit checks